### PR TITLE
Feature/37 transport react to vanilla js

### PR DESCRIPTION
**Pull request by massahito**

### CODE REVIEW

File: vanilla-pong-game/srcs/main.js [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+			component: GameResultComponent,
+			state: {}
+		}
+	],
+	{
+		playersInfo: {},
+		gameResults: [],
+	});
+
+router.gonextPage(location.pathname);
```
Comment (ShotaTanemura) [inline vanilla-pong-game/srcs/main.js]: Fix: `gonextPage` -> `goNextPage`

File: vanilla-pong-game/srcs/components/HomeComponent.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
+import {Component} from '../core/component.js'
+import { UserNameForm } from './UserNameForm.js';
+
+export class HomeComponent extends Component {
+	constructor(router, parameters, state) {
+		super(router, parameters, state);
+			let useNameForm = new UserNameForm(router, parameters, state);
```
Comment (ShotaTanemura) [inline vanilla-pong-game/srcs/components/HomeComponent.js]: typo: `useNameForm` -> `userNameForm`

File: vanilla-pong-game/srcs/components/UserNameForm.js [hunk 3]
```diff
@@ -0,0 +1,75 @@
+import {Component} from '../core/component.js'
+import { UserNameInput } from './UserNameInput.js';
+
+export class UserNameForm extends Component {
+	constructor(router, parameters, state) {
+		super(router, parameters, state);
+		this.num_of_user = 2;
```
Comment (ShotaTanemura) [inline vanilla-pong-game/srcs/components/UserNameForm.js]: `num_of_user` -> `numOfUser`

File: vanilla-pong-game/srcs/components/Tournament.js [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+
+export class Tournament extends Component {
+  constructor (route, parameters, state){
+    super(route, parameters, state);
+    this.playersInfo = this.getRouteContext("playersInfo");
+    this.gameResults = this.getRouteContext("gameResults");
+    this.rounds = []
+    this.displayTournament();
+    this.bracket = new Bracket(this.rounds, route, parameters, state);
+    this.element.insertBefore(this.bracket.element, this.gameStartButton);
```
Comment (ShotaTanemura) [inline vanilla-pong-game/srcs/components/Tournament.js]: fix: `insertBefore` -> `appendChild`

File: vanilla-pong-game/srcs/main.js (hunk 1 above)
Comment (ghost) [reply vanilla-pong-game/srcs/main.js]: resolved [a4dd9d8](https://github.com/ShotaTanemura/ft_transcendence/pull/38/commits/a4dd9d8ae355bbc68f1f54d6a4e9f77ec2b0d153)

File: vanilla-pong-game/srcs/components/HomeComponent.js (hunk 2 above)
Comment (ghost) [reply vanilla-pong-game/srcs/components/HomeComponent.js]: resolved [a4dd9d8](https://github.com/ShotaTanemura/ft_transcendence/pull/38/commits/a4dd9d8ae355bbc68f1f54d6a4e9f77ec2b0d153)

File: vanilla-pong-game/srcs/components/UserNameForm.js (hunk 3 above)
Comment (ghost) [reply vanilla-pong-game/srcs/components/UserNameForm.js]: resolved [a4dd9d8](https://github.com/ShotaTanemura/ft_transcendence/pull/38/commits/a4dd9d8ae355bbc68f1f54d6a4e9f77ec2b0d153)

File: vanilla-pong-game/srcs/components/Tournament.js (hunk 4 above)
Comment (ghost) [reply vanilla-pong-game/srcs/components/Tournament.js]: resolved [a4dd9d8](https://github.com/ShotaTanemura/ft_transcendence/pull/38/commits/a4dd9d8ae355bbc68f1f54d6a4e9f77ec2b0d153)

**Review by soooku — APPROVED** [review]
ありがとう！！

**Review by ShotaTanemura — APPROVED** [review]
LGTM

**Pull request merged**